### PR TITLE
Add vApp network functions

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Add new function need for handling vApp networks:

    Add AddNetwork, AddNetworkAsync
    Add AddOrgNetwork, AddOrgNetworkAsync
    Add UpdateNetwork, UpdateNetworkAsync
    Add UpdateOrgNetwork, UpdateOrgNetworkAsync
    Add RemoveNetwork
    Add GetUuidFromHref

### CODE REVIEW

File: govcd/vapp.go [hunk 1]
```diff
@@ -786,6 +795,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+		IsDeployed: false,
+	}
+	networkConfigurations = append(networkConfigurations,
+		vappConfiguration)
+
+	return updateNetworkConfigurations(vapp, networkConfigurations)
+
+}
+
+// UpdateNetwork updates vApp networks(isolated or connected to org VDC network)
```
Comment (lvirbalas) [inline govcd/vapp.go]:
```suggestion
// UpdateNetwork updates vApp networks (isolated or connected to Org VDC network)
```

File: govcd/vapp.go [hunk 2]
```diff
@@ -786,6 +795,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// UpdateNetworkAsync asyncronously updates vApp networks(isolated or connected to org VDC network).
```
Comment (lvirbalas) [inline govcd/vapp.go]:
```suggestion
// UpdateNetworkAsync asyncronously updates vApp networks (isolated or connected to Org VDC network).
```

File: govcd/vapp.go [hunk 3]
```diff
@@ -786,6 +795,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+		networkToUpdate.Configuration.Features.DhcpService.MaxLeaseTime = networkSettingsToUpdate.DhcpSettings.MaxLeaseTime
+		networkToUpdate.Configuration.Features.DhcpService.IPRange = networkSettingsToUpdate.DhcpSettings.IPRange
+	}
+
+	currentNetworkConfiguration.NetworkConfig[networkToUpdateIndex] = networkToUpdate
+
+	return updateNetworkConfigurations(vapp, currentNetworkConfiguration.NetworkConfig)
+}
+
+// UpdateOrgNetwork updates to vApp added org VDC network
```
Comment (lvirbalas) [inline govcd/vapp.go]: Maybe this would be better:
```suggestion
// UpdateOrgNetwork updates Org VDC network which is part of a vApp
```

File: govcd/vapp.go [hunk 4]
```diff
@@ -786,6 +795,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// UpdateOrgNetworkAsync asyncronously updates to vApp added org VDC network
```
Comment (lvirbalas) [inline govcd/vapp.go]: Maybe this:
```suggestion
// UpdateOrgNetworkAsync asyncronously updates Org VDC network which is part of a vApp
```

File: govcd/vapp.go [hunk 5]
```diff
@@ -814,7 +1145,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
 	return nil
 }
 
+// RemoveNetwork removes any network from vApp
```
Comment (lvirbalas) [inline govcd/vapp.go]: Did you mean this:
```suggestion
// RemoveNetwork removes any network (be it isolated or connected to an Org Network) from vApp
```

File: govcd/vapp.go [hunk 6]
```diff
@@ -814,7 +1145,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
...
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// RemoveNetworkAsync asyncronously removes vApp any network
```
Comment (lvirbalas) [inline govcd/vapp.go]: Sentence doesn't make sense.

File: govcd/vapp.go [hunk 7]
```diff
@@ -814,7 +1145,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
...
+// Accepts network Id or name
```
Comment (lvirbalas) [inline govcd/vapp.go]:
```suggestion
// Accepts network ID or name
```

File: govcd/vapp.go [hunk 8]
```diff
@@ -814,7 +1145,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
...
+func (vapp *VApp) RemoveNetworkAsync(id string) (Task, error) {
+
+	if id == "" {
+		return Task{}, fmt.Errorf("network id/name can't be empty")
```
Comment (lvirbalas) [inline govcd/vapp.go]:
```suggestion
		return Task{}, fmt.Errorf("network ID/name can't be empty")
```

File: govcd/vapp_test.go [hunk 9]
```diff
@@ -769,3 +769,356 @@ func (vcd *TestVCD) Test_GetVM(check *C) {
...
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp was not successfully created at setup")
+	}
+
+	if vcd.config.VCD.Network.Net1 == "" {
+		check.Skip("Skipping test because no network was given")
+	}
+
+	// Add Metadata
+	networkName := "AddAndRemoveVappNetworkTest"
```
Comment (lvirbalas) [inline govcd/vapp_test.go]: What about changing the `networkName` so they are unique for each of the new tests? They could have a word to depict the type of the network too.

File: govcd/vapp.go (hunk 1 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp.go (hunk 2 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp.go (hunk 3 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp.go (hunk 4 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp.go (hunk 5 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp.go (hunk 6 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp.go (hunk 7 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp.go (hunk 8 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: govcd/vapp_test.go (hunk 9 above)
Comment (vbauzysvmware) [reply govcd/vapp_test.go]: changed

File: govcd/system.go [hunk 10]
```diff
@@ -921,3 +921,24 @@ func (vcdClient *VCDClient) GetAdminOrgByNameOrId(identifier string) (*AdminOrg,
 	}
 	return entity.(*AdminOrg), err
 }
+
+// Returns the UUID part of an HREF
+// Similar to getBareEntityUuid, but tailored to HREF
+func GetUuidFromHref(href string) (string, error) {
```
Comment (Didainius) [inline govcd/system.go]: A few questions - do we really need this to be exported? Maybe it could work internally only?
Also - is there any different between our current private function `getUuidFromHref`? Can they be wrapped or reused?

File: govcd/vapp.go [hunk 11]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+	if err != nil {
+		return Task{}, err
+	}
+
+	// for case when range is one ip address
+	if newNetworkSettings.DhcpSettings != nil && newNetworkSettings.DhcpSettings.IPRange != nil && newNetworkSettings.DhcpSettings.IPRange.EndAddress == "" {
+		newNetworkSettings.DhcpSettings.IPRange.EndAddress = newNetworkSettings.DhcpSettings.IPRange.StartAddress
+	}
+
+	// explicitly check if to add data, to not send any values
```
Comment (Didainius) [inline govcd/vapp.go]: Could you rephrase, this one is hard to follow.

File: CHANGELOG.md [hunk 12]
```diff
@@ -8,6 +8,8 @@
 * Added method `catalog.GetVappTemplateByHref`, and expose methods `vdc.GetEdgeGatewayByHref` and `vdc.GetEdgeGatewayRecordsType`
 * Added methods `adminOrg.CreateOrgVdc`, `adminOrg.CreateOrgVdcAsync` and improved existing to support Flex VDC model. These new methods are dynamic as they change invocation behind the scenes based on vCD version [#285](https://github.com/vmware/go-vcloud-director/pull/285) 
 * Deprecated functions `adminOrg.CreateVdc` and `adminOrg.CreateVdcWait` [#285](https://github.com/vmware/go-vcloud-director/pull/285)
+* Added methods `vapp.AddNetwork`, `vapp.AddNetworkAsync`, `vapp.AddOrgNetwork`, `vapp.AddOrgNetworkAsync`, `vapp.UpdateNetwork`, `vapp.UpdateNetworkAsync`, `vapp.UpdateOrgNetwork`, `vapp.UpdateOrgNetworkAsync`, `vapp.RemoveNetwork`, `vapp.RemoveNetworkAsync` and `GetUuidFromHref` [#289](https://github.com/vmware/go-vcloud-director/pull/290)
+* Deprecated function `vapp.RemoveIsolatedNetwork` [#289](https://github.com/vmware/go-vcloud-director/pull/290)
```
Comment (Didainius) [inline CHANGELOG.md]: As far as I see this is not the only deprecated function. At least `vapp.AddIsolatedNetwork`, `AddRAWNetworkConfig` are also deprecated.

File: govcd/vapp.go [hunk 13]
```diff
@@ -814,7 +1148,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
...
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// RemoveNetworkAsync asyncronously removes any network (be it isolated or connected to an Org Network) from vApp
+// Accepts network ID or name
+func (vapp *VApp) RemoveNetworkAsync(id string) (Task, error) {
```
Comment (Didainius) [inline govcd/vapp.go]: I believe we use such variable name when name and id are accepted.

```suggestion
func (vapp *VApp) RemoveNetworkAsync(identifier string) (Task, error) {
```

File: govcd/vapp.go [hunk 14]
```diff
@@ -35,15 +35,21 @@ func (vcdCli *VCDClient) NewVApp(client *Client) VApp {
...
-	Name             string
-	Gateway          string
-	NetMask          string
-	DNS1             string
-	DNS2             string
-	DNSSuffix        string
-	GuestVLANAllowed *bool
-	StaticIPRanges   []*types.IPRange
-	DhcpSettings     *DhcpSettings
+	Id                 string
```
Comment (Didainius) [inline govcd/vapp.go]: I think we are using capital cased ID strings in types.

```suggestion
	ID                 string
```

File: govcd/vapp.go [hunk 15]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+// If orgNetwork is nil, then isolated network created.
+func (vapp *VApp) AddNetworkAsync(newNetworkSettings *VappNetworkSettings, orgNetwork *types.OrgVDCNetwork) (Task, error) {
+
+	err := validateNetworkConfigSettings(newNetworkSettings)
+	if err != nil {
+		return Task{}, err
+	}
+
+	// for case when range is one ip address
+	if newNetworkSettings.DhcpSettings != nil && newNetworkSettings.DhcpSettings.IPRange != nil && newNetworkSettings.DhcpSettings.IPRange.EndAddress == "" {
```
Comment (Didainius) [inline govcd/vapp.go]: Is there some specific scenario when it is specified like that? As far as I have seen ranges usually have start and end addresses populated.

File: govcd/vapp.go [hunk 16]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+		}
+	}
+
+	if newNetworkSettings.FirewallEnabled != nil {
+		if networkFeatures == nil {
+			networkFeatures = &types.NetworkFeatures{}
+		}
+		networkFeatures.FirewallService = &types.FirewallService{IsEnabled: *newNetworkSettings.FirewallEnabled}
+	}
+	if newNetworkSettings.FirewallEnabled != nil && newNetworkSettings.NatEnabled != nil {
```
Comment (Didainius) [inline govcd/vapp.go]: Does it imply that firewall must be enabled so that NAT rules can be enabled? This looks to be handled a bit differently in `AddOrgNetworkAsync`

File: govcd/vapp.go [hunk 17]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+
+	// for case when range is one ip address
+	if newNetworkSettings.DhcpSettings != nil && newNetworkSettings.DhcpSettings.IPRange != nil && newNetworkSettings.DhcpSettings.IPRange.EndAddress == "" {
+		newNetworkSettings.DhcpSettings.IPRange.EndAddress = newNetworkSettings.DhcpSettings.IPRange.StartAddress
+	}
+
+	// explicitly check if to add data, to not send any values
+	var networkFeatures *types.NetworkFeatures
+	if newNetworkSettings.DhcpSettings != nil {
+		networkFeatures = &types.NetworkFeatures{DhcpService: &types.DhcpService{
```
Comment (Didainius) [inline govcd/vapp.go]: This assignment looks like 1:1 copy. Maybe it could be directly assigned?

File: govcd/vapp.go [hunk 18]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+	if newNetworkSettings.FirewallEnabled != nil && newNetworkSettings.NatEnabled != nil {
+		networkFeatures.NatService = &types.NatService{IsEnabled: *newNetworkSettings.NatEnabled, NatType: "ipTranslation", Policy: "allowTrafficIn"}
+	}
+
+	networkConfigurations := vapp.VApp.NetworkConfigSection.NetworkConfig
+	vappConfiguration := types.VAppNetworkConfiguration{
+		NetworkName: newNetworkSettings.Name,
+		Description: newNetworkSettings.Description,
+		Configuration: &types.NetworkConfiguration{
+			FenceMode:        types.FenceModeNAT,
```
Comment (Didainius) [inline govcd/vapp.go]: Is it correct even when isolated network is being created?

File: govcd/vapp.go [hunk 19]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+			networkToUpdateIndex = index
+			break
+		}
+	}
+
+	if networkToUpdate == (types.VAppNetworkConfiguration{}) {
+		return Task{}, fmt.Errorf("not found network to update with Id %s", networkSettingsToUpdate.Id)
+	}
+
+	if networkToUpdate.Configuration.Features.NatService != nil && networkSettingsToUpdate.NatEnabled != nil {
```
Comment (Didainius) [inline govcd/vapp.go]: Would it be a big risk/breaking change to use `*bool` and assign directly instead of converting each value?

File: govcd/vapp.go [hunk 20]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+	if networkToUpdate.Configuration.ParentNetwork != nil && orgNetwork != nil && networkToUpdate.Configuration.ParentNetwork.HREF != orgNetwork.HREF {
+		networkToUpdate.Configuration.ParentNetwork = &types.Reference{HREF: orgNetwork.HREF}
+	}
+	if networkToUpdate.Configuration.ParentNetwork != nil && orgNetwork == nil {
+		networkToUpdate.Configuration.ParentNetwork = nil
+	}
+	networkToUpdate.Description = networkSettingsToUpdate.Description
+	networkToUpdate.NetworkName = networkSettingsToUpdate.Name
+	networkToUpdate.Configuration.GuestVlanAllowed = networkSettingsToUpdate.GuestVLANAllowed
+	networkToUpdate.Configuration.IPScopes.IPScope[0].Gateway = networkSettingsToUpdate.Gateway
```
Comment (Didainius) [inline govcd/vapp.go]: Are we not binding ourselves to first object while many IP ranges can exist?

File: govcd/vapp.go [hunk 21]
```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+	networkToUpdate.Configuration.IPScopes.IPScope[0].Netmask = networkSettingsToUpdate.NetMask
+	networkToUpdate.Configuration.IPScopes.IPScope[0].DNS1 = networkSettingsToUpdate.DNS1
+	networkToUpdate.Configuration.IPScopes.IPScope[0].DNS2 = networkSettingsToUpdate.DNS2
+	networkToUpdate.Configuration.IPScopes.IPScope[0].DNSSuffix = networkSettingsToUpdate.DNSSuffix
+	networkToUpdate.Configuration.IPScopes.IPScope[0].IPRanges = &types.IPRanges{IPRange: networkSettingsToUpdate.StaticIPRanges}
+
+	// for case when range is one ip address
+	if networkSettingsToUpdate.DhcpSettings != nil && networkSettingsToUpdate.DhcpSettings.IPRange != nil && networkSettingsToUpdate.DhcpSettings.IPRange.EndAddress == "" {
```
Comment (Didainius) [inline govcd/vapp.go]: Similar question to the one in other place - are there cases when only one field is specified?

File: govcd/system.go (hunk 10 above)
Comment (vbauzysvmware) [reply govcd/system.go]: This functions used widely in terraform also due that we need get ID from href -> specific vapp network handling. Private function is slight different - regular expression is slight changed.

File: govcd/vapp.go (hunk 11 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: updated

File: CHANGELOG.md (hunk 12 above)
Comment (vbauzysvmware) [reply CHANGELOG.md]: Thank you. Added

File: govcd/vapp.go (hunk 13 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: changed

File: govcd/vapp.go (hunk 14 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: changed

File: govcd/vapp.go (hunk 15 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: It's according how you provide data. `newNetworkSettings` allows you provide such way of data

File: govcd/vapp.go (hunk 16 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: made to be the same in AddOrgNetworkAsync

File: govcd/vapp.go (hunk 17 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: Types are different DhcpService and DhcpSettings

File: govcd/vapp.go (hunk 19 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: It used by other functions - which will be impacted, needs to be verified, etc.

File: govcd/vapp.go (hunk 20 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: Well, in UI you can't have more gateways, etc. So if you know how to verify need of that when we consider changing a lot this PR.

File: govcd/vapp.go (hunk 21 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: There is possibility of that.

File: govcd/system.go (hunk 10 above)
Comment (dataclouder) [reply govcd/system.go]: The only difference between the regular expressions in public and private functions is that the public one allows text after the UUID.
I think that, instead of adding a function that is 99% the same as another, we should add a parameter to match  either open ended or closed strings.
Alternatively, the two functions should have a different name, because `getUuidFromHref` matches an entity UUID (ID at the end), while `GetUuidFromHref` matches an UUID from an action HREF (ID/action at the end). 

Given that the action is discarded, I suggest this interface:

```go
func GetUuidFromHref(href string, idAtEnd bool) (string, error) {
   searchExpression := `^https://.+/([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})`
   if idAtEnd {
      searchExpression +=`$`
   } else {
      searchExpression += `.*$`
   }
   reGetID := regexp.MustCompile(searchExpression)
   // [...]
} 
```
This way, we use just one function for two needs.

Calls to `getUuidFromHref` will become `GetUuidFromHref(href, true)`, while calls to `GetUuidFromHref` will become `GetUuidFromHref(href, false)`

Comment (lvirbalas) [reply govcd/system.go]: I'm in favor of a common function as well to avoid duplication.

File: govcd/vapp.go [hunk 22]
```diff
@@ -786,6 +798,329 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
 
 }
 
+// AddNetwork creates isolated or nat routed(connected to Org VDC network) network for vApp.
```
Comment (dataclouder) [inline govcd/vapp.go]: This function (and the related `Async` function) need a better name.
In both `AddNetwork` and `AddOrgNetwork` we are passing an `OrgVdcNetwork` as a parameter, but in one case we are assigning an existing network to the vApp, and in the other we are creating a vApp network. 
I think the name in this case should be `CreateVappNetwork` or `AddVappNetwork` to avoid confusion

File: govcd/vapp.go [hunk 23]
```diff
@@ -814,7 +1149,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
...
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// RemoveNetworkAsync asyncronously removes any network (be it isolated or connected to an Org Network) from vApp
```
Comment (dataclouder) [inline govcd/vapp.go]:
```suggestion
// RemoveNetworkAsync asynchronously removes any network (be it isolated or connected to an Org Network) from vApp
```

File: govcd/vapp.go [hunk 24]
```diff
@@ -786,6 +798,329 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
...
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// AddNetworkAsync creates asyncronously isolated or nat routed network for vApp. Returns Task or error
```
Comment (dataclouder) [inline govcd/vapp.go]:
```suggestion
// AddNetworkAsync creates asynchronously isolated or nat routed network for vApp. Returns Task or error
```

File: govcd/system.go (hunk 10 above)
Comment (vbauzysvmware) [reply govcd/system.go]: good idea, change added.

File: govcd/vapp.go (hunk 18 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: improved

File: govcd/vapp.go (hunk 22 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: @Didainius what is your vote for naming?

File: govcd/vapp.go (hunk 23 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: Thank you. updated

File: govcd/vapp.go (hunk 24 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: Thank you. updated

File: govcd/vapp.go (hunk 16 above)
Comment (Didainius) [reply govcd/vapp.go]: After a discussion we proved that NAT cannot be enabled without firewall being enabled.

File: govcd/vapp.go (hunk 19 above)
Comment (Didainius) [reply govcd/vapp.go]: Alright. Lets live with it.

File: govcd/vapp.go [hunk 25]
```diff
@@ -814,7 +1149,57 @@ func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
 	return nil
 }
 
+// RemoveNetwork removes any network (be it isolated or connected to an Org Network) from vApp
+// Returns pointer to types.NetworkConfigSection or error
+func (vapp *VApp) RemoveNetwork(id string) (*types.NetworkConfigSection, error) {
```
[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply govcd/vapp.go]: renamed

File: CHANGELOG.md [hunk 26]
```diff
@@ -11,11 +11,12 @@
 * Added methods `EdgeGateway.GetAllNsxvDhcpLeases()`, `EdgeGateway.GetNsxvActiveDhcpLeaseByMac()`
   `VM.WaitForDhcpIpByNicIndexes()`, `VM.GetParentVApp()`, `VM.GetParentVdc()`
   [#283](https://github.com/vmware/go-vcloud-director/pull/283)
+* Added methods `vapp.AddNetwork`, `vapp.AddNetworkAsync`, `vapp.AddOrgNetwork`, `vapp.AddOrgNetworkAsync`, `vapp.UpdateNetwork`, `vapp.UpdateNetworkAsync`, `vapp.UpdateOrgNetwork`, `vapp.UpdateOrgNetworkAsync`, `vapp.RemoveNetwork`, `vapp.RemoveNetworkAsync` and `GetUuidFromHref` [#289](https://github.com/vmware/go-vcloud-director/pull/290)
+* Deprecated function `vapp.RemoveIsolatedNetwork`, `vapp.AddRAWNetworkConfig` and `vapp.AddIsolatedNetwork`  [#289](https://github.com/vmware/go-vcloud-director/pull/290)
```
[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply CHANGELOG.md]: fixed

File: govcd/vapp.go (hunk 22 above)
Comment (vbauzysvmware) [reply govcd/vapp.go]: renamed
